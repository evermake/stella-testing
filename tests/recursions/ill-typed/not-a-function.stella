language core;

extend with #fixpoint-combinator;

fn main(n : Bool) -> Nat {
  return fix(n)
}
