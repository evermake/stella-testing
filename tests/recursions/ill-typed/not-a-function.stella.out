An error occurred during typechecking!
===================
Primary type error:
===================
Type Error Tag: [ERROR_UNEXPECTED_TYPE_FOR_EXPRESSION]
expected type
  fn (Nat) -> Nat
but got
  Bool
for expression
  n
========================
Alternative type errors:
========================
Type Error Tag: [ERROR_NOT_A_FUNCTION]
expected a one-argument function type but got
  Bool
for the expression
  n
in the expression
  fix (n)